### ISSUE TITLE
Automatic Database Update not working

### ISSUE THREAD

**Issue by goodmorty** [ampache/ampache#1071]
On build 3.8.2-develop with database version 380002 I am getting this request:

Version: 3.8 Build:0003
- Add unique constraint on tag_map table.

However, when I click update nothing happens, it just reloads the page. I initially had two db update requests on the page and one seemed to go through the first time with this one remaining (I can't remember what the first one was).

**Comment by iamnumbersix** [comment]
Confirmed

**Comment by minnesotags** [comment]
It also hangs at the update page. Either that, or it is in progress of updating my 500+ album database. A progress bar would be quite helpful.

**Comment by Psy-Virus** [comment]
Hey ho,
this happens because of duplicated entries within the tag_map table. A workaround on my site was to truncate the tag_map folder and update my catalogs. But you also can manually find these duplicates. They're described within the ampache log.

Hope this helps.

**Comment by goodmorty** [comment]
Eli5

**Comment by Psy-Virus** [comment]
XD I had to google what "eli5" means...

If you use php my admin, head over to your database, search the tag_map table and click on empty (truncate in older versions).

if you only have console access, try: mysql -uusername -ppassword TRUNCATE ampachedatabasename.tag_map

the username and the password have to be written directly behind the switches. !!No space between!!

If you do not write anything behind the switches, you will be asked for you username and pw.

:-)

**Comment by Psy-Virus** [comment]
Oh and then you simply have to update or verify your Ampache catalogs. Either through the Web-UI (admin settings -> show catalogs -> catalog action: verify) or through the console:
`php /Ampachebasedir/bin/catalog_update.inc -v`

**Comment by goodmorty** [comment]
Thank you very much!

**Comment by Psy-Virus** [comment]
:sweat_smile: great, thought I've to make screen shots :stuck_out_tongue_winking_eye:

You're welcome :smiley:

**Comment by minnesotags** [comment]
That's a manual workaround, not a fix.

**Comment by goodmorty** [comment]
But the issue was still solved

**Comment by Psy-Virus** [comment]
Well the duplicate entries could be ignored, but I think this could bring up other issues. Maybe we should wait until @Afterster looked into this issue.

As I see it, the work isn't worth it to code this serverside. Maybe the update description should be more detailed so every user directly sees what he or she have to do while the update page hangs.

But as I said, let's wait for @Afterster ^^

**Comment by Afterster** [comment]
The problem is duplicate entries. I changed the sql update to automatically remove duplicates, that should fix the issue.

**Comment by minnesotags** [comment]
I had experienced many duplicate entries previously which i had to manually delete. This makes sense.

**Comment by phirestalker** [comment]
Thanks for the quick fix.
Stupid question, how do I delete a library? They are all empty as of now.
